### ISSUE TITLE
Make PMD configuration scope application level instead of project level

### ISSUE THREAD

**Issue by jborgers** [amitdev/PMD-Intellij#197]
As a IntelliJ user, I would like to configure custom rules, enable skip test sources, target JDK, statistics URL, Threads and rules annotated in Editor for IntelliJ, for all projects, instead of for each project separately, so that I don't have to do this again for each project.
Right now there is a complex and flaky merging mechanism between projects to merge custom rules, and only that. It should be properly solved.

**Comment by adangel** [comment]
I'm not sure, if this "global sharing between projects feature" is still needed, given that #182 / #100 is done now: Sharing configuration between project members of the same project works now by committing the file `.idea/PMDPlugin.xml`.

The use case for sharing configuration between _different_ projects sounds dubious to me... Different projects have different needs and most likely different rulesets...

Not having such a feature makes the plugin so much simpler...

**Comment by jborgers** [comment]
Right, so it is a matter of choosing the 'right' level. And this may differ among users.
When I add a custom rule set and I enable 'skip test sources' and I work on one project, then I load another project, I expect the custom rules set to still be available and test sources still to be skipped. In my view this is configuration on global level. And I know only of others who expect the same, and they are surprised that configuration has changed for the other project (no custom rules, all defaults.)

But there may be other use cases. Maybe custom rules for frameworks used only in one project, or between different languages (currently only Java is supported.)

**Comment by jborgers** [comment]
Good to have the links. It might be an option to define globally and override on project level if needed. And indeed not make it into a mess.

**Comment by jborgers** [comment]
Related: #144
